start.sh

```diff
@@ -53,33 +48,28 @@ docker-compose --compatibility up -d grafana
 echo "Sleeping for $SLEEP_TIME secs to give Grafana some time to start up..."
 sleep $SLEEP_TIME
 
-existingPerfanaUserId=$(curl -X GET -H "Content-Type: application/json"  http://$GRAFANA_CREDS@localhost:3000/api/auth/keys 2>/dev/null | jq '.[] | select(.name == "Perfana")' | jq .id)
-
-if [ -z "$existingPerfanaUserId" ];  then
-    echo "Creating Grafana API key..."
-    apiKey=$(curl -X POST -H "Content-Type: application/json" -d '{"name":"Perfana","role":"Admin"}' http://$GRAFANA_CREDS@localhost:3000/api/auth/keys 2>/dev/null | jq -r '.key')
-    if [[ -z "$apiKey" ]] || [[ "$apiKey" == "null" ]]; then
-        echo "Issue creating Grafana API key, no API key returned from create call: abort!"
-        exit 123
-    fi
-    echo "Replacing apiKey in docker-compose file with: ${apiKey}"
-    if [ "$(check-os)" == "mac" ]; then
-        sed -i '' "s/\"apiKey\": \".*\"/\"apiKey\": \"$apiKey\"/g" docker-compose.yml
-    else
-        sed -i "s/\"apiKey\": \".*\"/\"apiKey\": \"$apiKey\"/g" docker-compose.yml
-    fi
+perfanaUser=`curl -X GET -H "Content-Type: application/json" "http://$GRAFANA_CREDS@localhost:3000/api/auth/keys" 2>/dev/null`
+echo "perfanaUser: $perfanaUser"
+if [[ $perfanaUser != *"Perfana"* || $perfanaUser != *"Admin"* ]]; then
+  echo "No Perfana user found. Creating new Grafana API key."
+  API_KEY=$(curl -X POST -H "Content-Type: application/json" -d '{"name":"Perfana","role":"Admin"}' http://$GRAFANA_CREDS@localhost:3000/api/auth/keys 2>/dev/null | sed "s/.*key\"\:\"\(.*\)\"}/\1/")
+  if [[ -z "$API_KEY" ]] || [[ "$API_KEY" == "null" ]]; then
+    echo "Issue creating Grafana API key, no API key returned from create call: abort!"
+    exit 123
+  fi
+  echo "Replacing apiKey in docker-compose file with: ${API_KEY}"
+  sed "s/\"apiKey\": \".*\"/\"apiKey\": \"$API_KEY\"/g" docker-compose-template.yml > docker-compose.yml
```

Comment:
moet check doen of dit ook allemaal goed werkt op de mac, met inline replace was syntax anders, zal nu misschien wel goed gaan

Reply:
The lines that I saw were identical so I figured yes...